### PR TITLE
Faster validation scripts

### PR DESCRIPTION
**Pull request by dnephin**
Extracted from #27359 

The `hack/make` framework of creating bundle directories and requiring execution of scripts to be run from a driver (hack/make.sh) is ok for build tasks that produce artifacts, but has little to no benefit for these validation/check scripts. This PR changes them to source all their dependencies directly, so they no longer require the hack/make.sh driver. It also moves them from `hack/make` to `hack/validate` so that it's clear they can be run directly without the driver.

They also run as a single process now, so instead of every one of them having to perform a git fetch, hack/validate/all does a single git fetch, reducing the overall runtime of validation.

### CODE REVIEW

**Review by thaJeztah — CHANGES_REQUESTED** [review]
@dnephin some files changed permissions; can you check if that was intentional?

**Comment by thaJeztah** [comment]
ping @tianon @vdemeester PTAL :heart:

File: hack/validate/.validate [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
+set -e -o pipefail
+
 if [ -z "$VALIDATE_UPSTREAM" ]; then
 	# this is kind of an expensive check, so let's not do this twice if we
 	# are running more than one validate bundlescript
 
 	VALIDATE_REPO='https://github.com/docker/docker.git'
 	VALIDATE_BRANCH='master'
 
-	if [ "$TRAVIS" = 'true' -a "$TRAVIS_PULL_REQUEST" != 'false' ]; then
```
Comment (vdemeester) [inline hack/validate/.validate]: :heart:

File: Makefile [hunk 2]
```diff
@@ -133,7 +133,7 @@ tgz: build ## build the archives (.zip on windows and .tgz\notherwise) containin
 	$(DOCKER_RUN_DOCKER) hack/make.sh dynbinary binary cross tgz
 
 validate: build ## validate DCO, Seccomp profile generation, gofmt,\n./pkg/ isolation, golint, tests, tomls, go vet and vendor
-	$(DOCKER_RUN_DOCKER) hack/make.sh validate-dco validate-default-seccomp validate-gofmt validate-pkg validate-lint validate-test validate-toml validate-vet validate-vendor
+	$(DOCKER_RUN_DOCKER) bash -c 'hack/validate/all && hack/validate/vendor'
```
Comment (vdemeester) [inline Makefile]: Shouldn't `all` be rename default and having a `all` that runs `default` and `vendor ? 👼

**Review by vdemeester — COMMENTED** [review]
LGTM 🐸

**Review by thaJeztah — APPROVED** [review]
nevermind, answering my earlier question; see why they're changed :-) "approving" to neutralize my earlier "request changes"

File: Makefile (hunk 2 above)
Comment (dnephin) [inline Makefile]: hmm, that could work too

Comment (tianon) [inline Makefile]: I agree that `all` is a bit misleading if it doesn't include `vendor` :sweat_smile:

**Review by tianon — COMMENTED** [review]
This is excellent! :heart: :heart:

**Comment by jhowardmsft** [comment]
LGTM. @dnephin Can you open an issue and assign to me once this is merged to cleanup the Windows CI scripts and remove the workarounds. Thanks.

**Comment by thaJeztah** [comment]
@dnephin can you rename `all`, per https://github.com/docker/docker/pull/27964#discussion_r86219119, then I think we're ready to go

**Comment by bfirsh** [comment]
Ahah, didn't notice #27964! I'm happy to wait and fix this up after that merges.

**Comment by dnephin** [comment]
Cool, I moved `all` -> `default` and created a new `all` that is really everything (for `make validate`).

Also rebased

**Review by vdemeester — APPROVED** [review]
Still and even more LGTM 👼

**Review by tianon — APPROVED** [review]

**Pull request merged**